### DISCUSSION TITLE
Validation on Blur & isValid | isDirty on Type

### DISCUSSION THREAD

**Discussion by laurenzhonauer** [react-hook-form/react-hook-form#7573] [Q&A] [answered]
## Scenario

I have a form with react-hook-form in place. 
There are X Input fields and a submit button.
Below each input field is a div, displaying any validation errors regarding that field.

Validation is simply the input has to be longer than 3 chars

As soon as the whole form has state isValid or !isDirty, the submit button is activated.

## Behaviour when mode: "onBlur"

When i leave focus on any input fields, error messages may appear. I have to type into all input fields **and then exit focus on the last field** before the submit button is activated.

## Behaviour when mode: "all" or "onChange"

As soon as I type a char in any of the inputs, the error appears because i do not yet have entered 4 characters.
If I entered enough characters into every input the submit button is active **and i do not have to exit focus on the last input**

## My desired Behaviour

I want errors regarding an input field on blur and at the same time i want the submit button to be active as early as possible. That means I do not want to have to blur the last field. Button should be active as soon as the last field has 4 characters in it.

Is there a feasible way to achive this with the library? Any help is highly appreceated 😄

**Comment by Moshyfawn** [comment, marked as answer]
It seems like maybe you want validation to happen `onChange`, rather than `onBlur`, to be able to track the current form state of `isValid`. There's[ a `delayError: number` prop](https://react-hook-form.com/api/useform#useFormRef) on `useForm`. You can kinda debounce the value change to see the user's gonna keep typing or left the field with an invalid value. See [this CSB for demonstration](https://codesandbox.io/s/delayed-validation-example-uw4rp?file=/src/App.tsx:441-451)

**Reply by laurenzhonauer** [reply]
Yes, im pretty sure this is as far as my requirements are going to fulfilled.

Thanks for the quick answer!

**Reply by laurenzhonauer** [reply]
> 

My answer reads so pessimistic. This is **exactly** what i was looking for. Thanks again! ❤️

**Reply by Moshyfawn** [reply]
No problem!
